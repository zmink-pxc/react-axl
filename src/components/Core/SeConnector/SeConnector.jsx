import React from 'react';
import SeTerminalGroup from '@bit/zmink.axioline.smart-elements.se-terminal-group';
import styles from './SeConnector.module.scss';

export default function SeConnector(props){
    return (<div className={styles.root}>
        <div className={styles.shadow}></div>
        <SeTerminalGroup key='tg-1' left={true} />
        <SeTerminalGroup key='tg-2' />
        <SeTerminalGroup key='tg-3' left={true} />
        <SeTerminalGroup key='tg-4' />
        <SeTerminalGroup key='tg-5' left={true} />
        <SeTerminalGroup key='tg-6' />
        <SeTerminalGroup key='tg-7' left={true} />
        <SeTerminalGroup key='tg-8' />
    </div>)
}