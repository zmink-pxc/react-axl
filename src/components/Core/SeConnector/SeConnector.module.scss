@import '@styles/_axioline.scss';

.root {
    position: relative;
    padding-top: .2mm;
    width: 9mm;
    height: 29mm;
    background-color: $axioline-grey;
}

.shadow{
    z-index: 2;
    position: absolute;
    width:  100%;
    height: 100%;
    top: 0px;
    left: 0px;
    box-shadow: inset 0mm 0mm 2mm 0px rgba(0, 0, 0, 0.75);
}