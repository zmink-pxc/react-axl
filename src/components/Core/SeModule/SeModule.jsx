import React from 'react';
import PropTypes from 'prop-types';
import styles from './SeModule.module.scss';
import SeLed from '@core/SeLed/SeLed.jsx';
import SeLock from '@core/SeLock/SeLock.jsx';

export default function SeModule(props){
    const {partName,leds,ledLabels,children,...other} = props;
    return (<div className={styles.root} {...other}>
        <div className={styles.partName}><div style={{marginLeft: '1mm'}}>{partName}</div></div>
        <div className={styles.mainContent}>
            <LedArea ledLabels={ledLabels} ledProps={leds}/><div className={styles.conArea}>{children}</div>
        </div>
        <div className={styles.partName}><div style={{marginLeft: '1mm'}}>{partName}</div></div>
    </div>)
}

SeModule.propTypes = {
    partNumber: PropTypes.string.isRequired,
    partName:   PropTypes.string.isRequired,
    terminals:  PropTypes.object
}

SeModule.defaultProps = {
    ledLabels: ['E','01','02','03','04','05','06','07','08','09','10','11','12','13','14','15','16','E']
}

function LedArea(props){
    return (<div className={styles.ledArea}>{
        props.ledLabels.map((label,index)=>{
            const bigLed = (index === 0) || (index === 17);
            return (index === 9) ? (<SeLock/>):(<div className={styles.ledContainer}><SeLed isErrLed={bigLed}/><div>{label}</div></div>)
        })
    }</div>)
    
}