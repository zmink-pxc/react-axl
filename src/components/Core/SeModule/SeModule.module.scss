@import '@styles/_axioline.scss';

.root {
    outline: 0.1mm solid black;
    font-size: 2mm;
    width: 14.9mm;
    height: 62.2mm;
    background-color: $axioline-grey;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
}

.partName {
    font-size: 1mm;
    height: 5mm;
    display: flex;
    flex-direction: row;
    align-items: center;
    width: 100%;
    padding-left: 4mm;
}

.mainContent {
    width: 100%;
    height: 58mm;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
}

.conArea{
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    height: 58mm;
}

.ledArea{
    height: 58mm;
    width: 5mm;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: flex-start;
}

.ledContainer{
    width:  4mm;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
}