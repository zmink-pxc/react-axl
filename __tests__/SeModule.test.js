import React from 'react';
import SeModule from '@core/SeModule/SeModule.jsx';
import SeConnector from '@core/SeConnector/SeConnector.jsx';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  var component = renderer.create(<SeModule/>)
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

    component = renderer.create(<SeModule partName={'SeTest'} partNumber={'1111111'} ledLabels={['E','','','','','','','','','','','','','','','','','E']} ><SeConnector/></SeModule>);

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(<SeModule partName={'SeTest'} partNumber={'1111111'} ledLabels={['E','','','','','','','','','','','','','','','','','E']} ><SeConnector isBlack={true}/></SeModule>);

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
   
});