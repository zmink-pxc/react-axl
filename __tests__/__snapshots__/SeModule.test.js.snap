// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="root"
>
  <div
    className="partName"
  >
    <div
      style={
        Object {
          "marginLeft": "1mm",
        }
      }
    />
  </div>
  <div
    className="mainContent"
  >
    <div
      className="ledArea"
    >
      <div
        className="ledContainer"
      >
        <div
          className="led_base errLed"
        />
        <div>
          E
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          01
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          02
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          03
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          04
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          05
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          06
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          07
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          08
        </div>
      </div>
      <div
        className="root"
      >
        <div
          className="indent"
        />
        <div
          className="indent"
        />
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          10
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          11
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          12
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          13
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          14
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          15
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          16
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base errLed"
        />
        <div>
          E
        </div>
      </div>
    </div>
    <div
      className="conArea"
    />
  </div>
  <div
    className="partName"
  >
    <div
      style={
        Object {
          "marginLeft": "1mm",
        }
      }
    />
  </div>
</div>
`;

exports[`renders correctly 2`] = `
<div
  className="root"
>
  <div
    className="partName"
  >
    <div
      style={
        Object {
          "marginLeft": "1mm",
        }
      }
    >
      SeTest
    </div>
  </div>
  <div
    className="mainContent"
  >
    <div
      className="ledArea"
    >
      <div
        className="ledContainer"
      >
        <div
          className="led_base errLed"
        />
        <div>
          E
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="root"
      >
        <div
          className="indent"
        />
        <div
          className="indent"
        />
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base errLed"
        />
        <div>
          E
        </div>
      </div>
    </div>
    <div
      className="conArea"
    >
      <div
        className="root"
      >
        <div
          className="shadow"
        />
        <div
          className="root"
        >
          <div
            className="container"
          >
            <div
              className="edge edge_end"
            >
              <div
                className="inset inset_end"
              >
                <div
                  className="center_circle"
                >
                  <div
                    className="terminal_center"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="plunger_base"
          >
            <div
              className="middle"
            />
          </div>
        </div>
        <div
          className="root"
        >
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="plunger_base"
          >
            <div
              className="middle"
            />
          </div>
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="container"
          >
            <div
              className="edge"
            >
              <div
                className="inset"
              >
                <div
                  className="center_circle"
                >
                  <div
                    className="terminal_center"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="root"
        >
          <div
            className="container"
          >
            <div
              className="edge edge_end"
            >
              <div
                className="inset inset_end"
              >
                <div
                  className="center_circle"
                >
                  <div
                    className="terminal_center"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="plunger_base"
          >
            <div
              className="middle"
            />
          </div>
        </div>
        <div
          className="root"
        >
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="plunger_base"
          >
            <div
              className="middle"
            />
          </div>
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="container"
          >
            <div
              className="edge"
            >
              <div
                className="inset"
              >
                <div
                  className="center_circle"
                >
                  <div
                    className="terminal_center"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="root"
        >
          <div
            className="container"
          >
            <div
              className="edge edge_end"
            >
              <div
                className="inset inset_end"
              >
                <div
                  className="center_circle"
                >
                  <div
                    className="terminal_center"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="plunger_base"
          >
            <div
              className="middle"
            />
          </div>
        </div>
        <div
          className="root"
        >
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="plunger_base"
          >
            <div
              className="middle"
            />
          </div>
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="container"
          >
            <div
              className="edge"
            >
              <div
                className="inset"
              >
                <div
                  className="center_circle"
                >
                  <div
                    className="terminal_center"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="root"
        >
          <div
            className="container"
          >
            <div
              className="edge edge_end"
            >
              <div
                className="inset inset_end"
              >
                <div
                  className="center_circle"
                >
                  <div
                    className="terminal_center"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="plunger_base"
          >
            <div
              className="middle"
            />
          </div>
        </div>
        <div
          className="root"
        >
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="plunger_base"
          >
            <div
              className="middle"
            />
          </div>
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="container"
          >
            <div
              className="edge"
            >
              <div
                className="inset"
              >
                <div
                  className="center_circle"
                >
                  <div
                    className="terminal_center"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="partName"
  >
    <div
      style={
        Object {
          "marginLeft": "1mm",
        }
      }
    >
      SeTest
    </div>
  </div>
</div>
`;

exports[`renders correctly 3`] = `
<div
  className="root"
>
  <div
    className="partName"
  >
    <div
      style={
        Object {
          "marginLeft": "1mm",
        }
      }
    >
      SeTest
    </div>
  </div>
  <div
    className="mainContent"
  >
    <div
      className="ledArea"
    >
      <div
        className="ledContainer"
      >
        <div
          className="led_base errLed"
        />
        <div>
          E
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="root"
      >
        <div
          className="indent"
        />
        <div
          className="indent"
        />
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base"
        />
        <div>
          
        </div>
      </div>
      <div
        className="ledContainer"
      >
        <div
          className="led_base errLed"
        />
        <div>
          E
        </div>
      </div>
    </div>
    <div
      className="conArea"
    >
      <div
        className="root"
      >
        <div
          className="shadow"
        />
        <div
          className="root"
        >
          <div
            className="container"
          >
            <div
              className="edge edge_end"
            >
              <div
                className="inset inset_end"
              >
                <div
                  className="center_circle"
                >
                  <div
                    className="terminal_center"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="plunger_base"
          >
            <div
              className="middle"
            />
          </div>
        </div>
        <div
          className="root"
        >
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="plunger_base"
          >
            <div
              className="middle"
            />
          </div>
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="container"
          >
            <div
              className="edge"
            >
              <div
                className="inset"
              >
                <div
                  className="center_circle"
                >
                  <div
                    className="terminal_center"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="root"
        >
          <div
            className="container"
          >
            <div
              className="edge edge_end"
            >
              <div
                className="inset inset_end"
              >
                <div
                  className="center_circle"
                >
                  <div
                    className="terminal_center"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="plunger_base"
          >
            <div
              className="middle"
            />
          </div>
        </div>
        <div
          className="root"
        >
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="plunger_base"
          >
            <div
              className="middle"
            />
          </div>
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="container"
          >
            <div
              className="edge"
            >
              <div
                className="inset"
              >
                <div
                  className="center_circle"
                >
                  <div
                    className="terminal_center"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="root"
        >
          <div
            className="container"
          >
            <div
              className="edge edge_end"
            >
              <div
                className="inset inset_end"
              >
                <div
                  className="center_circle"
                >
                  <div
                    className="terminal_center"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="plunger_base"
          >
            <div
              className="middle"
            />
          </div>
        </div>
        <div
          className="root"
        >
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="plunger_base"
          >
            <div
              className="middle"
            />
          </div>
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="container"
          >
            <div
              className="edge"
            >
              <div
                className="inset"
              >
                <div
                  className="center_circle"
                >
                  <div
                    className="terminal_center"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="root"
        >
          <div
            className="container"
          >
            <div
              className="edge edge_end"
            >
              <div
                className="inset inset_end"
              >
                <div
                  className="center_circle"
                >
                  <div
                    className="terminal_center"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="plunger_base"
          >
            <div
              className="middle"
            />
          </div>
        </div>
        <div
          className="root"
        >
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="plunger_base"
          >
            <div
              className="middle"
            />
          </div>
          <div
            style={
              Object {
                "width": "0.5mm",
              }
            }
          />
          <div
            className="container"
          >
            <div
              className="edge"
            >
              <div
                className="inset"
              >
                <div
                  className="center_circle"
                >
                  <div
                    className="terminal_center"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="partName"
  >
    <div
      style={
        Object {
          "marginLeft": "1mm",
        }
      }
    >
      SeTest
    </div>
  </div>
</div>
`;
